controller/spawn/local.go

```diff
@@ -0,0 +1,134 @@
+package spawn
+
+import (
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+	"sync"
+
+	"github.com/google/uuid"
+)
+
+// Start dealbot daemons locally
+type LocalSpawner struct {
+	sync.Mutex
+	endpoint string
+	cmds     map[string]exec.Cmd
+}
+
+func (s *LocalSpawner) Spawn(d *Daemon) error {
+	log.Infow("spawining daemon", "region", d.Region)
+	exe, err := os.Executable()
+	if err != nil {
+		return err
+	}
+	d.Id = uuid.New().String()
+	data_dir, err := ioutil.TempDir("/tmp", d.Id)
+	if err != nil {
+		return err
+	}
+	cmd := exec.Cmd{
+		Path: exe,
+		Env: envStr(map[string]string{
+			"DEALBOT_ID":                  d.Id,
+			"DEALBOT_DATA_DIRECTORY":      data_dir,
+			"DEALBOT_TAGS":                strings.Join(d.Tags, ","),
+			"DEALBOT_WALLET_ADDRESS":      d.Wallet,
+			"DEALBOT_WORKERS":             strconv.Itoa(d.Workers),
+			"DEALBOT_CONTROLLER_ENDPOINT": s.endpoint,
+			"DEALBOT_MIN_FIL":             strconv.Itoa(d.MinFil),
+			"DEALBOT_MIN_CAP":             strconv.Itoa(d.MinCap),
+		}),
+		Args:   []string{exe, "daemon"},
+		Stdout: os.Stdout,
+		Stderr: os.Stderr,
+	}
+	log.Infow("starting daemon with cli", "command", cmd.String())
+	err = cmd.Start()
+	if err != nil {
+		log.Info("failed to start daemon", "id", d.Id, "err", err)
+		return err
+	}
+	s.Lock()
+	s.cmds[d.Id] = cmd
+	s.Unlock()
+	go func() {
+		err := cmd.Wait()
+		if err != nil {
+			log.Errorw("dealbot daemon crashed", "id", d.Id, "err", err)
+		}
+		log.Info("dealbot exited normally", "id", d.Id, "code", cmd.ProcessState.ExitCode())
+		s.Lock()
+		delete(s.cmds, d.Id)
+		s.Unlock()
+	}()
+	return nil
+}
+
+func (s *LocalSpawner) Get(regionid string, daemonid string) *Daemon {
+	daemoncmd, ok := s.cmds[daemonid]
```

Comment:
In other places, `s.Lock()` / `s.Unlock()` is used to protect access to `s.cmds`.  Should it be used here also?